gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-apikey/src/main/java/io/gravitee/gateway/security/apikey/ApiKeyAuthenticationHandler.java

```diff
@@ -132,6 +157,39 @@ private String readApiKey(Request request) {
         return apiKey;
     }
 
+    private List<String> getCustomApiKeyHeadersFromPlans() {
+        if (api != null && api.getPlans() != null) {
+            return api
+                .getPlans()
+                .stream()
+                .filter(Plan::isApiKey)
+                .filter(plan -> StringUtils.hasText(plan.getSecurityDefinition()))
+                .map(plan -> {
+                    try {
+                        JsonNode securityDef = objectMapper.readTree(plan.getSecurityDefinition());
+                        if (
+                            securityDef != null &&
+                            securityDef.has(ENABLE_CUSTOM_API_KEY_HEADER_FIELD) &&
+                            securityDef.has(CUSTOM_API_KEY_HEADER_FIELD)
+                        ) {
+                            if (securityDef.get(ENABLE_CUSTOM_API_KEY_HEADER_FIELD).asBoolean(false)) {
+                                String headerName = securityDef.get(CUSTOM_API_KEY_HEADER_FIELD).asText();
+                                if (StringUtils.hasText(headerName)) {
+                                    return headerName;
+                                }
+                            }
+                        }
```

Comment:
it can be simplified

```suggestion
                        if (
                            securityDef != null &&
                            ecurityDef.at(ENABLE_CUSTOM_API_KEY_HEADER_FIELD).asBoolean()
                        ) {
                            String headerName = securityDef.at(CUSTOM_API_KEY_HEADER_FIELD).asText();
                            if (StringUtils.hasText(headerName)) {
                                return headerName;
                            }
                        }
```

but to use JSON pointer tho path should be define like it


```java
    private static final String CUSTOM_API_KEY_HEADER_FIELD = "/apiKeyHeader";
    private static final String ENABLE_CUSTOM_API_KEY_HEADER_FIELD = "/enableCustomApiKeyHeader";
```